internal/executor/node/node_group.go

```diff
@@ -100,6 +101,10 @@ func (kubernetesNodeInfoService *KubernetesNodeInfoService) GetAllAvailableProce
 	return util2.FilterNodes(allNodes, kubernetesNodeInfoService.isAvailableProcessingNode), nil
 }
 
+func (kubernetesNodeInfoService *KubernetesNodeInfoService) GetAllProcessingNodes() ([]*v1.Node, error) {
```

Comment:
Possibly this should be "GetAllNodes"

Processing nodes implies we filter it in someway. 
 - i.e We'd include all with taints that are in kubernetesNodeInfoService.toleratedTaints